components/mfa/confirm/index.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {loadMe} from 'mattermost-redux/actions/users';
+import {getTeamMemberships} from 'mattermost-redux/selectors/entities/teams';
+
+import {isEmptyObject} from 'utils/utils';
+
+import Confirm from './confirm.jsx';
+
+function mapStateToProps(state) {
+    return {
+
+        // Assume we need to load the user if they don't have any team memberships loaded
+        shouldLoadUser: isEmptyObject(getTeamMemberships(state)),
```

Comment:
In the case that the user is truly not on any teams and has already loaded all their data (not sure how they'd get into that state) then this might cause an extra few HTTP requests to load the user's data again. In 99.9% of the cases though if their team memberships are empty, they likely need to load all their data.

Reply:
It's unfortunate that the `Confirm` component has to "know" about this. And I presume this fix might actually help in other invocations to `redirectUserToDefaultTeam`? What if we moved the empty test into that action instead?

Reply:
That's actually a really good idea. Let me try that 👍